### PR TITLE
Dev phlf

### PR DESCRIPTION
**Pull request by PHLF**
[Total ground war support repairing](https://github.com/MHeasell/rwe/projects/2#card-18281490)

As @MHeasell stated there is a lot of similarities between the build and the repair logic.

There is room for some code refactor in this PR especially in the UnitBehaviorService

### CODE REVIEW

File: src/rwe/Unit.cpp [hunk 1]
```diff
@@ -95,19 +77,29 @@ namespace rwe
...
         auto deltaEnergy = newProgressEnergy - oldProgressEnergy;
-        auto deltaMetal = newProgressMetal - oldProgressMetal;
+        auto deltaMetal  = newProgressMetal - oldProgressMetal;
 
         return BuildCostInfo{buildTimeContribution, deltaEnergy, deltaMetal};
     }
 
+    bool Unit::addRepairProgress(unsigned int repairTimeContribution)
+    {
+        if (buildTimeCompleted == buildTime)
```
Comment (PHLF) [inline src/rwe/Unit.cpp]: I hate this, but I was not able to find a cleaner solution fitting the current build logic.

File: src/rwe/Unit.cpp [hunk 2]
```diff
@@ -128,6 +120,7 @@ namespace rwe
         if (hitPoints + deltaHp >= maxHitPoints)
         {
             hitPoints = maxHitPoints;
+            lifeState = LifeState::Alive;
```
Comment (PHLF) [inline src/rwe/Unit.cpp]: I had to add this as repairing using the build logic could invalidate the "isBeingBuilt" check

File: src/rwe/UnitBehaviorService.cpp [hunk 3]
```diff
@@ -764,14 +763,137 @@ namespace rwe
         return false;
     }
 
+    bool UnitBehaviorService::handleRepairOrder(UnitId unitId, const RepairOrder& repairOrder)
```
Comment (PHLF) [inline src/rwe/UnitBehaviorService.cpp]: Mostly handleBuildOrder copy pasted. Lot of room for refactoring.

File: src/rwe/proto/serialization.cpp [hunk 4]
```diff
@@ -45,6 +45,11 @@ namespace rwe
             std::visit(visitor, o.target);
         }
 
+        void operator()(const RepairOrder& o){
+            auto& out = *cmd->mutable_repair();
```
Comment (PHLF) [inline src/rwe/proto/serialization.cpp]: I did not really understood what I was doing there

File: src/rwe/GameScene.cpp [hunk 5]
```diff
@@ -859,29 +903,32 @@ namespace rwe
...
+                auto worldPos
+                    = minimapToWorld
+                      * Vector3f(static_cast<float>(mousePos.x) + 0.5f, static_cast<float>(mousePos.y) + 0.5, 0.0f);
                 auto newCameraPos = cameraConstraint.clamp(Vector2f(worldPos.x, worldPos.z));
                 camera.setPosition(Vector3f(newCameraPos.x, camera.getRawPosition().y, newCameraPos.y));
             }
         }
 
-        hoveredUnit = getUnitUnderCursor();
+        elementOfInterest = getElementOfInterest();
```
Comment (PHLF) [inline src/rwe/GameScene.cpp]: Terrain position and units are both and the only elements the player can interacts with directly or indirectly, did I get it right?

**Comment by MHeasell** [comment]
Pierre-Henri, thank you very much for contributing!

I have had a brief look at this PR, but I'm afraid I cannot accept it in its current form. The PR is very large, and this seems to be because it makes modifications to `.clang-format` and makes a large number of formatting changes.

What is your reason for this? I am very happy with the current project code style -- any change to this needs to have a strong justification.

**Comment by PHLF** [comment]
Please excuse me about that as I do have the very bad habit with .clang-format to carelessly merge formatting changes with code changes.
I know this tends to make commit diff not much readable.
I will find the time to format back the code using the original .clang-format file you made for the project and make a cleaner PR.
I juste wanted you to be able to have a look at what was done and give dome feedback regarding the code logic.

Nevertheless, I encourage you to have a look at clang-format vertical alignement features as this  emphasizes the code logic and make dense code more readable.

Maybe I'll do a separate PR for this if you are interested.

**Pull request closed without merging**